### PR TITLE
Configurable GasLimit for op-deployer

### PR DESCRIPTION
**Pull request by blmalone**
Gas limit should be configurable via op-deployer.

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12228** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12228?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12224** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12224?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blmalone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: op-chain-ops/deployer/pipeline/opchain.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func DeployOPChain(ctx context.Context, env *Env, artifactsFS foundry.StatDirFs,
 		L2ChainId:              chainID.Big(),
 		OpcmProxy:              st.ImplementationsDeployment.OpcmProxyAddress,
 		SaltMixer:              st.Create2Salt.String(), // passing through salt generated at state initialization
+		GasLimit:               30_000_000,              // TODO: make this configurable
```
Comment (blmalone) [inline op-chain-ops/deployer/pipeline/opchain.go:43]: @mslipper Let me know if you think we should add more logic in this PR for this to be added to the Intent struct.

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol [hunk 2]
```diff
@@ -486,7 +487,7 @@ contract OPContractsManager is ISemver, Initializable {
                 _input.basefeeScalar,
                 _input.blobBasefeeScalar,
                 bytes32(uint256(uint160(_input.roles.batcher))), // batcherHash
-                30_000_000,
+                _input.gasLimit,
```
Comment (mds1) [inline packages/contracts-bedrock/src/L1/OPContractsManager.sol:490]: We need to make this change in `OPContractsManagerInterop` also

**Review by mds1 — APPROVED** [review]

File: packages/contracts-bedrock/src/L1/OPContractsManager.sol (hunk 2 above)
Comment (blmalone) [reply packages/contracts-bedrock/src/L1/OPContractsManager.sol:490]: good call: https://github.com/ethereum-optimism/optimism/pull/12224/commits/eef95031abc7d667a53379e90cef6cec01e7b03e

**Review by mslipper — APPROVED** [review]

**Pull request merged**